opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -691,8 +695,32 @@ namespace Opm {
     }
 
 
-    void EclipseGrid::fwriteEGRID( const std::string& filename, bool output_metric ) const {
-        ecl_grid_fwrite_EGRID( m_grid.get() , filename.c_str(), output_metric );
+
+
+    inline ert_ecl_unit_enum to_ert_unit( UnitSystem::UnitType t ) {
+        switch ( t ) {
+        case UnitSystem::UNIT_TYPE_METRIC: return ERT_ECL_METRIC_UNITS;
+        case UnitSystem::UNIT_TYPE_FIELD: return ERT_ECL_FIELD_UNITS;
+        case UnitSystem::UNIT_TYPE_LAB: return ERT_ECL_LAB_UNITS;
+        }
+
+        throw std::invalid_argument("unhandled enum value");
+    }
+
+
+    void EclipseGrid::fwriteEGRID( const std::string& filename, UnitSystem::UnitType unitSystem) const {
```

Comment:
Slightly confusing naming, considering it's named `system`, but you're sending the `type` tag.